File: ios/banner/AdropBannerViewManager.m
#import <React/RCTViewManager.h>
 
@interface RCT_EXTERN_MODULE(AdropBannerViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(unitId, NSString)
RCT_EXPORT_METHOD(load:(NSNumber *)reactTag){}

@end

File: ios/banner/AdropBannerViewManager.swift
import React
import AdropAds
import UIKit


@objc(AdropBannerViewManager)
class AdropBannerViewManager: RCTViewManager {
    
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    override func view() -> AdropBannerViewWrapper? {
        return AdropBannerViewWrapper(bridge: self.bridge)
    }
    
    @objc func load(_ reactTag: NSNumber) {
        DispatchQueue.main.async {
            if let uiView =  self.bridge.uiManager.view(forReactTag: reactTag),
               let banner = uiView as? AdropBannerViewWrapper {
                banner.load()
            }
        }
    }
}
